torch/csrc/jit/tensorexpr/schedule.cpp

```diff
@@ -399,6 +399,126 @@ class FunctionInliner : public IRMutator {
   std::unordered_set<const Var*> func_var_set_;
 };
 
+/**
+ * Actual algorithm for "inlining" of rand().  Bind variables to rand() calls.
+ */
+class RandomInliner : public IRMutator {
+ public:
+  RandomInliner(const std::unordered_set<Function*>& funcs) : funcs_(funcs) {
+    for (auto const* func : funcs_) {
+      func_vars_.insert(func->func_var(0));
+    }
+  }
+
+  static bool is_inner_loop(const For* loop) {
+    Block* body = dynamic_cast<Block*>(loop->body());
+    if (!body) {
+      return false;
+    }
+    if (body->nstmts() != 1) {
```

Comment:
This seems to be overly conservative. It should be allowed to have several statements in the body if neither of them is a For loop:
```
for // <-- loop
  block // <-- body
    store1 // <-- *body->stmts().begin()
    store2 // <-- *(body->stmts().begin()++)
```